src/sentry/static/sentry/app/views/projects/redirectSentry9Project.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import {withRouter} from 'react-router';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import LoadingError from 'app/components/loadingError';
+
+// TODO: This is react-router v4 <Redirect to="path/" /> component to allow things
+//       to be declarative
+export class Redirect extends React.Component {
+  static propTypes = {
+    router: PropTypes.object.isRequired,
+    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
+  };
+
+  componentDidMount() {
+    this.props.router.replace(this.props.to);
+  }
+
+  render() {
+    return null;
+  }
+}
+
+class ProjectDetailsInner extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: null,
+    project: null,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({
+      loading: true,
+      error: null,
+    });
+
+    const {orgId, projectId} = this.props;
+
+    try {
+      const projectRequest = this.props.api.requestPromise(
+        `/projects/${orgId}/${projectId}/`
+      );
+
+      const project = await projectRequest;
+      this.setState({
+        loading: false,
+        error: null,
+        project,
+      });
+    } catch (error) {
+      this.setState({
+        loading: false,
+        error,
+        project: null,
+      });
+    }
+  };
+
+  getProjectId = () => {
+    return _.get(this.state.project, 'id');
+  };
+
+  hasProjectId = () => {
+    const projectID = this.getProjectId(this.state.project);
+    return _.isString(projectID) && projectID.length > 0;
+  };
+
+  render() {
+    const childrenProps = {
+      ...this.state,
+      getProjectId: this.getProjectId,
+      hasProjectId: this.hasProjectId,
+    };
+
+    return this.props.children(childrenProps);
+  }
+}
+
+export const ProjectDetails = withRouter(withApi(ProjectDetailsInner));
+
+const redirectSentry9Project = generateRedirectRoute => {
+  class RedirectSentry9Project extends React.Component {
+    static propTypes = {
+      router: PropTypes.object.isRequired,
+
+      params: PropTypes.shape({
+        orgId: PropTypes.string.isRequired,
+        projectId: PropTypes.string.isRequired,
+      }).isRequired,
+    };
+
+    render() {
+      const {orgId, projectId} = this.props.params;
+
+      return (
+        <ProjectDetails orgId={orgId} projectId={projectId}>
+          {({loading, error, hasProjectId, getProjectId}) => {
+            if (loading) {
+              return null;
+            }
+
+            if (!hasProjectId()) {
+              if (_.get(error, 'status') === 404) {
+                return (
+                  <div className="container">
```

Comment:
Do we need this class name?

Reply:
I just lifted from https://github.com/getsentry/sentry/blob/851ddc79404fbdd698ff8efc75e513707dbad02d/src/sentry/static/sentry/app/views/projects/projectContext.jsx#L245-L249

I removed the `<div/>` wrapper.